epson-escpr-1.7.20 failed with exit code 2 after 11s in buildPhase

## Describe the bug

epson-escpr package for printer driver fails to build on nixos-unstable with log- 
```
 Making all in src
3609   │ make[2]: Entering directory '/build/epson-inkjet-printer-escpr-1.7.20/src'
3610   │ gcc -DHAVE_CONFIG_H -I. -I..    -I../lib -g -O2 -c -o epson_escpr-filter.o `test -f 'filter.c' || echo './'`filter.c
3611   │ filter.c: In function 'epsInitJob':
3612   │ filter.c:131:9: error: implicit declaration of function 'obsClear' [
       │ cit-function-declaration^G-Wimplicit-function-declaration]
3613   │   131 |         obsClear();
3614   │       |         ^~~~~~~~
3615   │ filter.c: In function 'main':
3616   │ filter.c:345:15: error: implicit declaration of function 'SetupJobAttrib' [
       │ -Wimplicit-function-declaration^G-Wimplicit-function-declaration]
3617   │   345 |         err = SetupJobAttrib(&jobAttr);
3618   │       |               ^~~~~~~~~~~~~~
3619   │ filter.c:354:15: error: implicit declaration of function 'pageAllocBuffer' [
       │ x-Wimplicit-function-declaration^G-Wimplicit-function-declaration]
3620   │   354 |         err = pageAllocBuffer();
3621   │       |               ^~~~~~~~~~~~~~~
3622   │ filter.c:373:15: error: implicit declaration of function 'SendStartJob'; did you mean 'epsStartJob'? [
       │ /Warning-Options.html#index-Wimplicit-function-declaration^G-Wimplicit-function-declaration]
3623   │   373 |         err = SendStartJob(FALSE);
3624   │       |               ^~~~~~~~~~~~
3625   │       |               epsStartJob
3626   │ filter.c:536:33: error: implicit declaration of function 'PrintBand'; did you mean 'epsPrintBand'? [
       │ arning-Options.html#index-Wimplicit-function-declaration^G-Wimplicit-function-declaration]
3627   │   536 |                                 PrintBand (rever_buf, bandBmp.widthBytes, &printHeight);
3628   │       |                                 ^~~~~~~~~
3629   │       |                                 epsPrintBand
3630   │ filter.c: In function 'set_pips_parameter':
3631   │ filter.c:832:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' [
       │ c/Warning-Options.html#index-Wno-unused-result^G-Wunused-result]
3632   │   832 |         read (STDIN_FILENO, &page_num, 1);
3633   │       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3634   │ make[2]: *** [Makefile:476: epson_escpr-filter.o] Error 1
3635   │ make[2]: Leaving directory '/build/epson-inkjet-printer-escpr-1.7.20/src'
3636   │ make[1]: *** [Makefile:445: all-recursive] Error 1
3637   │ make[1]: Leaving directory '/build/epson-inkjet-printer-escpr-1.7.20'
3638   │ make: *** [Makefile:372: all] Error 2
```

## Steps To Reproduce

Steps to reproduce the behavior:

1. Nix flake- 
```nix
  services = {
    printing.enable = true;
    printing.drivers = [pkgs.epson-escpr];
};
```
2. rebuild switch


**Entire Log file-** 
[epson.log](https://github.com/user-attachments/files/18246714/epson.log)

## Expected behavior

Nix build without errors

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241217.4989a24`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/9bdl6wa2qck3q0f7l892plk5lwy3pvx1-source`


## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@artuuge 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].



[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ep/epson-escpr/package.nix
+++ b/pkgs/by-name/ep/epson-escpr/package.nix
@@ -1,7 +1,18 @@
-{ lib, stdenv, fetchurl, cups }:
+{
+  lib,
+  stdenv,
+  fetchurl,
+  cups,
+  fetchpatch,
+  dos2unix,
+  automake115x,
+  autoconf,
+}:
 
-let version = "1.7.20";
-in stdenv.mkDerivation {
+let
+  version = "1.8.6-1";
+in
+stdenv.mkDerivation {
   pname = "epson-escpr";
   inherit version;
 
@@ -12,16 +23,46 @@ in stdenv.mkDerivation {
     # version.
     # NOTE: Don't forget to update the webarchive link too!
     urls = [
-      "https://download3.ebz.epson.net/dsc/f/03/00/13/76/45/5ac2ea8f9cf94a48abd64afd0f967f98c4fc24aa/epson-inkjet-printer-escpr-${version}-1lsb3.2.tar.gz"
+      "https://download3.ebz.epson.net/dsc/f/03/00/16/21/81/74d098a47c3a616713079c9cd5904b468bb33dea/epson-inkjet-printer-escpr-${version}.tar.gz"
 
-      "https://web.archive.org/web/https://download3.ebz.epson.net/dsc/f/03/00/13/76/45/5ac2ea8f9cf94a48abd64afd0f967f98c4fc24aa/epson-inkjet-printer-escpr-${version}-1lsb3.2.tar.gz"
+      "https://web.archive.org/web/https://download3.ebz.epson.net/dsc/f/03/00/16/21/81/74d098a47c3a616713079c9cd5904b468bb33dea/epson-inkjet-printer-escpr-${version}.tar.gz"
     ];
-    sha256 = "sha256:09rscpm557dgaflylr93wcwmyn6fnvr8nc77abwnq97r6hxwrkhk";
+    sha256 = "sha256-hVbX4OXPe4y37Szju3uVdXlVdjX4DFSN/A2Emz3eCcQ=";
   };
+  # DOS Line Endings in /lib directory and UNIX Line endings in /src directory.
+  # So convert everything to UNIX style line endings.
+  # Since the files are modified, autoconf and automake is required
+  postUnpack = ''
+    dir=''${src%-*}
+    cd ''${dir#*-}
+    local f
+    for f in $(find ./ -type f || die); do
+      ${dos2unix}/bin/dos2unix $f
+    done
+    cd ..
+  '';
 
-  patches = [ ./cups-filter-ppd-dirs.patch ];
+  patches = [
+    ./cups-filter-ppd-dirs.patch
+    (fetchpatch {
+      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-print/epson-inkjet-printer-escpr/files/epson-inkjet-printer-escpr-1.8-missing-include.patch";
+      hash = "sha256-L4WhaxPQnJYyqCH00wiKIlFNMmCEXoGe5d7hJ5TMyEI=";
+    })
+    (fetchpatch {
+      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/net-print/epson-inkjet-printer-escpr/files/1.8.6-warnings.patch";
+      hash = "sha256-wSY0LZv2b+1kF7TfPolt554g79y2Ce6N/JqgjJyd3Ag=";
+    })
+  ];
 
+  # To suppress error (Stripping trailing CRs from patch; use --binary to disable.)
+  patchFlags = [
+    "-p1"
+  ];
   buildInputs = [ cups ];
+  nativeBuildInputs = [
+    automake115x
+    autoconf
+  ];
 
   meta = with lib; {
     homepage = "http://download.ebz.epson.net/dsc/search/01/search/";

